iip-12.md

```diff
@@ -0,0 +1,323 @@
+```
+IIP: 12
+Title: Support staking action via JSON-RPC
+Author: Haixiang Liu(haixiang@iotex.io), Dustin Xie (dustin.xie@iotex.io)
+Status: Draft
+Type: Standards Track
+Category: Core
+Created: 2021-08-23
+```
+
+## Abstract
+
+This proposal explains the mechanism of supporting staking action via JSON-RPC 
+
+## Motivation
+
+With the growth of eco-system of IoTeX, the demands for staking via JSON-RPC is increasing. Currently, staking-related features are avilable in IoTeX `Antenna` API, which sends native staking actions to the API nodes via gRPC protocol. However, transaction transmission via JRPC(ethereum-based) is incompatible with our staking transaction transmission via gRPC. This is because the data structure of native ethereum transaction only contains six data fields(`AccountNonce`, `Price`, `GasLimit`, `Recipient`, `Amount`, `Payload`), whereas our staking actions includes some extra data fields(e.g. `staking candidate`, `staking amount`, etc.). So it is hard to send our native staking action directly via JSON-RPC.
+
+## Specification
+
+The discussion will be divided into two part: 
+
+1. How are staking actions sent to IoTeX blockchain via JSON-RPC?
+
+2. How are staking actions via JSON-RPC verified?
+
+### Part I: Transaction Transportation
```

Comment:
- use either Transaction or Action persistently 
- Is Transportation the right term?

Reply:
Encode Native Staking Action To Web3.js Staking Action  ?

Reply:
> * use either Transaction or Action persistently
> * Is Transportation the right term?

Transportation -> Transmission?